homeassistant/components/fitbit/sensor.py

```diff
@@ -360,6 +364,11 @@ def __init__(
         self.extra = extra
         if self.extra is not None:
             self._attr_name = f"{self.extra.get('deviceVersion')} Battery"
+            self._attr_unique_id = (
```

Comment:
I think it might make more sense to define a default, and then append battery-specific suffix:
```python
self._attr_unique_id = f"{user_profile['encodedId']}_{description.key}"
if self.extra is not None::
    self._attr_name = f"{self.extra.get('deviceVersion')} Battery"
    self._attr_unique_id = f"{self._attr_unique_id}_{self.extra.get('id')}_battery"
```

Reply:
Good idea, updated

Reply:
Can you confirm what it looks like for the battery sensors now?
Maybe adjust the description of the PR with the new unique id.

Reply:
Sure I've updated the description. Here's the new battery IDs:

```
7NHXM8_devices/battery_1472464222_battery
7NHXM8_devices/battery_FX0081613702081719090664204C925A17_battery
```

Reply:
Probably no need for the trailing `_battery` in that case

Reply:
I thought that might be the case... that's why I wanted to check!
Yes please remove

Reply:
Done!